tests/container_tests/test_zip_container.py

```diff
@@ -173,192 +183,188 @@ def test_open_non_exist(self):
         with self.fs_handler.open_as_container(non_exist_file) as handler:
             self.assertRaises(IOError, handler.open, non_exist_file)
 
-    def test_open_non_normalized_path(self):
-        cases = [
-            # not normalized path
-            {"path_or_prefix": '././testdir2//../testdir2/testfile1',
-             "expected": self.test_string}]
-        for case in cases:
-            with self.fs_handler.open_as_container(
-                    os.path.abspath(self.zip_file_path)) as handler:
-                with handler.open(case['path_or_prefix'], "r") as zipped_file:
-                    self.assertEqual(case['expected'], zipped_file.read())
-
-    def test_list(self):
+    @parameterized.expand([
+        # not normalized path
+        ['././{}//../{}/{}'.format(ZIP_TEST_FILENAME_LIST["dir_name2"],
+                                   ZIP_TEST_FILENAME_LIST["dir_name2"],
+                                   ZIP_TEST_FILENAME_LIST["zipped_file_name"])]
+    ])
+    def test_open_non_normalized_path(self, path_or_prefix):
+        with self.fs_handler.open_as_container(
+                os.path.abspath(self.zip_file_path)) as handler:
+            with handler.open(path_or_prefix, "r") as zipped_file:
+                self.assertEqual(self.test_string, zipped_file.read())
+
+    @parameterized.expand([
+        # default case get the first level from the root
+        ["",
+         [ZIP_TEST_FILENAME_LIST["dir_name1"].rstrip('/'),
```

Comment:
Why do you use 'rstrip('/')' ?

Reply:
These are some old code when the dirname ends with "/". I will remove them